runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -328,8 +439,27 @@ void chpl_mem_layerInit(void) {
 
 
 void chpl_mem_layerExit(void) {
-  if (heap.base != NULL) {
-    // ignore errors, we're exiting anyways
-    pthread_mutex_destroy(&heap.alloc_lock);
+  int hpi;
+
+  for (hpi = 0; hpi < get_num_heaps(); hpi++) {
+    if (heaps[hpi].base != NULL) {
+      // ignore errors, we're exiting anyways
+      (void) pthread_mutex_destroy(&heaps[hpi].alloc_lock);
+    }
   }
 }
+
+
+bool chpl_mem_impl_alloc_localizes(void) {
```

Comment:
I don't see any particular reason to make it chpl_bool since it isn't shared across the interface with Chapel (and I think in too many places in the runtime we use chpl_bool where we don't need it).  That said, I'm not averse to changing it if you'd prefer that.